### PR TITLE
Delete empty file without confirmation

### LINKED ISSUE #693: Enhancement: Don't prompt/confirm when deleting empty files
This is one subtle, but really nice behavior of Sublime. When creating files sometimes I get the name wrong and, for whatever reason, I'd rather delete the file and re-create than just rename it. Atom adds friction by prompting "Are you sure you want to delete the selected item?" – this gets old. There's no real consequence for deleting an empty file, so the prompt would be great to remove. Adjusting the behavior would also be great when removing files you don't end up needing after scaffolding a project.

TLDR: the prompt gets to be a bit much.

(moved from https://github.com/atom/atom/issues/10313)

### PR DESCRIPTION
**Pull request by m133225**
Fixes #693

Simple change to delete without prompting when the only item being deleted is an empty file.

### CODE REVIEW

File: lib/tree-view.coffee [hunk 1]
```diff
@@ -546,23 +546,32 @@ class TreeView extends View
           buttons: ['OK']
         return
 
+    if selectedPaths.length is 1 and fs.isFileSync(selectedPaths[0])
+      if fs.getSizeSync(selectedPaths[0]) is 0
+        @moveToTrash(selectedPaths)
+        return
+
+    moveToTrash = @moveToTrash
```
Comment (thomasjo) [inline lib/tree-view.coffee]: Not necessary. Make the arrow in the `atom.confirm` button callback below a "fat arrow" to capture the `this` scope, then delete this line.

File: lib/tree-view.coffee [hunk 2]
```diff
@@ -546,23 +546,32 @@ class TreeView extends View
...
     atom.confirm
       message: "Are you sure you want to delete the selected #{if selectedPaths.length > 1 then 'items' else 'item'}?"
       detailedMessage: "You are deleting:\n#{selectedPaths.join('\n')}"
       buttons:
         "Move to Trash": ->
```
Comment (thomasjo) [inline lib/tree-view.coffee]: Make this a fat arrow; `=>` to avoid the unnecessary `moveToTrash` variable.

File: lib/tree-view.coffee [hunk 3]
```diff
@@ -546,23 +546,32 @@ class TreeView extends View
...
-          if failedDeletions.length > 0
-            atom.notifications.addError "The following #{if failedDeletions.length > 1 then 'files' else 'file'} couldn't be moved to trash#{if process.platform is 'linux' then " (is `gvfs-trash` installed?)" else ""}",
-              detail: "#{failedDeletions.join('\n')}"
-              dismissable: true
+          moveToTrash(selectedPaths)
         "Cancel": null
 
+  moveToTrash: (pathsToDel) ->
+    failedDeletions = []
+    for pathToDel in pathsToDel
```
Comment (thomasjo) [inline lib/tree-view.coffee]: Rename these variables to `pathToDelete` and `pathsToDelete`.

**Comment by m133225** [comment]
@thomasjo Made the suggested changes!

**Comment by jerone** [comment]
Should empty files that are under source control be excluded from this feature?

**Comment by lee-dohm** [comment]
Upon further review, we would rather not change this behavior. Thanks very much for the effort that was put into this PR.

**Pull request closed without merging**